tests/samplers_tests/tpe_tests/test_multi_objective_sampler.py

```diff
@@ -345,11 +312,11 @@ def value_fn(idx: int) -> float:
     ) as mock2:
         mock1.return_value = attrs.value
         mock2.return_value = attrs.value
-        discrete_uniform_suggestion = sampler.sample_independent(
-            study, trial, "param-a", disc_dist
-        )
-    assert 1.0 <= discrete_uniform_suggestion <= 100.0
-    assert abs(int(discrete_uniform_suggestion * 10) - discrete_uniform_suggestion * 10) < 1e-3
+        suggestion = sampler.sample_independent(study, trial, "param-a", dist)
+    if float_dist.log or float_dist.step != 1.0:
```

Comment:
I think this condition is always `True` given the parametrization.

Furthermore, I'm not sure `float_suggeestion` would be the same as `suggestion` even if the parameterization is expected. (the current code only checks if they are not identical (https://github.com/optuna/optuna/pull/3337/files#diff-e52a24e2c14ee4ff0d8b5e62f38c4e515ec3dad12446570f0771ef74d27689faL314)). Does `sample_independent` return the same suggestion only by re-initializing `TPESampler` with reseed?

Reply:
Thank you so much for pointing out, I have fixed the conditioning branch to 
`if float_dist.log or float_dist.step == 1.0`
and re-seeded.